src/rest.cpp

```diff
@@ -590,6 +590,51 @@ static bool rest_chaininfo(const std::any& context, HTTPRequest* req, const std:
     }
 }
 
+RPCHelpMan getdeploymentinfo();
+
+static bool rest_deploymentinfo(const std::any& context, HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req)) return false;
+
+    std::string hashStr;
+    const RESTResponseFormat rf = ParseDataFormat(hashStr, strURIPart);
+
+    switch (rf) {
+    case RESTResponseFormat::JSON: {
+        JSONRPCRequest jsonRequest;
+        jsonRequest.context = context;
+        jsonRequest.params = UniValue(UniValue::VARR);
+
+        if (!hashStr.empty()) {
+            uint256 hash;
+            if (!ParseHashStr(hashStr, hash)) {
+                return RESTERR(req, HTTP_BAD_REQUEST, "Invalid hash: " + hashStr);
+            }
+
+            const uint256 hash_bi{ParseHashV(hashStr, "blockhash")};
+            const ChainstateManager* chainman = GetChainman(context, req);
+            if (!chainman) return false;
+            if (!WITH_LOCK(::cs_main, return chainman->m_blockman.LookupBlockIndex(hash_bi))) {
```

Comment:
`hash_bi` is an unnecessary temporary (if you want to keep it, would just name it `hash`)

```diff
-            const uint256 hash_bi{ParseHashV(hashStr, "blockhash")};
             const ChainstateManager* chainman = GetChainman(context, req);
             if (!chainman) return false;
-            if (!WITH_LOCK(::cs_main, return chainman->m_blockman.LookupBlockIndex(hash_bi))) {
+            if (!WITH_LOCK(::cs_main, return chainman->m_blockman.LookupBlockIndex(ParseHashV(hashStr, "blockhash")))) {
                 return RESTERR(req, HTTP_BAD_REQUEST, "Block not found");
             }
```

Reply:
Addressed it because makes the code cleaner. nice one!